src/net/sourceforge/kolmafia/session/InventoryManager.java

```diff
@@ -1733,14 +1733,18 @@ public static void checkVampireVintnerWine() {
     InventoryManager.checkItemDescription(ItemPool.VAMPIRE_VINTNER_WINE);
   }
 
-  public static void checkCoatOfPaint() {
+  public static void checkCoatOfPaint(boolean playerClassChanged) {
     AdventureResult COAT_OF_PAINT = ItemPool.get(ItemPool.COAT_OF_PAINT, 1);
-    String mod = Preferences.getString("_coatOfPaintModifier");
+
     if (!KoLCharacter.hasEquipped(COAT_OF_PAINT, EquipmentManager.SHIRT)
-        && !KoLConstants.inventory.contains(COAT_OF_PAINT)) {
+        && COAT_OF_PAINT.getCount(KoLConstants.inventory) == 0
+        && COAT_OF_PAINT.getCount(KoLConstants.closet) == 0) {
```

Comment:
`InventoryManager.getAccessibleCount` instead (or one of the related methods)? Or do we really want exactly equipped, inventory + closet?